### ISSUE TITLE
Bamboo-mutex-plugin tends to block the Bamboo server on a regular basis

### ISSUE THREAD

**Issue by mekugi** [ampedandwired/bamboo-mutex-plugin#1]
Hello Charles,

Thanks for sharing your quite useful plugin.

Readme reads that plans may get stuck on very rare occasions. In our experience, it seems to get stuck very regularly ( >= once a day).
I wonder if it could be related to the Bamboo usage: we have ca. 200 plans running with up to 10 plans sharing the same mutex key.
Said 10 plans are regularly launched at the same time depending on the CM submissions.
When it get stucks, build queue is empty and stuck plans are shown with an hourglass. Clicking the stop button on one of the plans interrupts the Java sleep and the plan starts running.
But the main issue is that it seems to prevent the Bamboo server from polling the CM repositories also for plans that do not use any mutexes. I am no Bamboo expert, but could the polling on a few concurrent plans actually block all the threads of the Bamboo server?

Configuration:
Bamboo 4.3.3 build 3403
Bamboo-mutex-plugin 1.0

Best regards

**Comment by ampedandwired** [comment]
Yes, each thread that's waiting for a mutex consumes a Bamboo worker thread, so yes I guess it's possible to run out of threads if you have enough of these running. You might be able to edit the Bamboo or servlet container config to increase the number of threads.

I never ran this plugin with so many jobs, so I assume you're seeing the deadlock problem more frequently because of this. I'll have a look when I get some time to see if there's a way to avoid these deadlocks.
